rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/activityList.spec.ts

```diff
@@ -0,0 +1,443 @@
+      }
+      await bulkDeleteButton.trigger("click");
+      console.log("Triggered bulk delete button click");
+
+      // Select and click checkboxes for deletion (Component line: Line No: 253 - 266 )
+      const checkboxes = wrapper.findAll(
+        '[data-testid="bulk-delete-checkbox"]',
+      );
+      console.log("Number of checkboxes:", checkboxes.length);
+      checkboxes.forEach((checkbox, index) => {
+        const inputElement = checkbox.element as HTMLInputElement;
+        console.log(`Checkbox ${index} value:`, inputElement.value);
+      });
+      await checkboxes.at(0)?.trigger("click");
+      console.log("First checkbox checked");
+      await checkboxes.at(1)?.trigger("click");
+      console.log("Second checkbox checked");
+
+      // Mock updated reports list after deletion
+      axiosMock.get.mockImplementationOnce((url) => {
+        if (url.includes("eventsAjax")) {
+          console.log("Fetch of reports after deletion");
+          // Assuming 2 reports were deleted, adjust the slice accordingly
+          return Promise.resolve({
+            data: {
+              total: reports.length - 2,
+              offset: 0,
+              auth: { projectAdmin: true, deleteExec: true },
+              bulkEditMode: false,
+              showBulkDelete: true,
+              reports: reports.slice(2),
+            },
+          });
+        }
+
+        return Promise.resolve({ data: {} });
+      });
+      // Trigger delete confirmation (Component line:79-85)
+      const confirmDeleteButton = wrapper.find(
+        '[data-testid="delete-selected-executions"]',
+      );
+      if (
+        !confirmDeleteButton.exists() ||
+        (confirmDeleteButton.element as HTMLButtonElement).disabled
+      ) {
+        console.error("Confirm delete button not found in DOM or is disabled");
+        return;
+      }
+      await confirmDeleteButton.trigger("click");
+      await wrapper.vm.$nextTick();
+      console.log("Triggered confirm delete button click");
+      // Ensure modal is rendered Line 146-152
+      const modal = wrapper.findComponent({ ref: "bulkexecdeleteresult" });
+      expect(modal.exists()).toBe(true);
+      if (modal.exists()) {
+        console.log("Modal HTML:", modal.html());
+      } else {
+        console.error("Modal not found in the DOM.");
+      }
+      // Check if modal contains the confirm delete button Line 164 -168
+      const modalConfirmDeleteButton = wrapper.find(
+        '[data-testid="confirm-delete"]',
+      );
+      expect(modalConfirmDeleteButton.exists()).toBe(true);
+      console.log(
+        "Modal confirm delete button exists:",
+        modalConfirmDeleteButton.exists(),
+      );
+      if (!modalConfirmDeleteButton.exists()) {
+        console.error("Modal confirm delete button not found in DOM");
+        return;
+      }
+      // Trigger modal confirm delete button click
+      await modalConfirmDeleteButton.trigger("click");
+      await wrapper.vm.$nextTick();
```

Comment:
After modalConfirmDeleteButton is clicked, it will call the method `bulkDeleteExecutions`, it is necessary to mock `executionBulkDelete` method as well.